glean-core/src/upload/mod.rs

```diff
@@ -656,14 +680,27 @@ impl PingUploadManager {
     ) -> UploadTaskAction {
         use UploadResult::*;
 
+        let stop_time = time::precise_time_ns();
+
         if let Some(label) = status.get_label() {
             let metric = self.upload_metrics.ping_upload_failure.get(label);
             metric.add_sync(glean, 1);
         }
 
+        let send_ids = {
+            let mut lock = self.in_flight.write().unwrap();
+            lock.remove(document_id)
+        };
+
         match status {
             HttpStatus { code } if (200..=299).contains(&code) => {
                 log::info!("Ping {} successfully sent {}.", document_id, code);
+                if let Some((success_id, failure_id)) = send_ids {
+                    self.upload_metrics
+                        .sending_success
+                        .set_stop_and_accumulate(glean, success_id, stop_time);
+                    self.upload_metrics.sending_failure.cancel_sync(failure_id);
+                }
```

Comment:
Suggestion: If `send_ids.is_none()` then that means 🤔  we've removed this doc_id before we got here, or we never inserted it in the first place... either way, that's an inconsistency. Perhaps we should instrument this?

Reply:
+1 from me on instrumenting this (or filing a bug to instrument it)

Reply:
Instrument the instrumentation 🤔 I'm all for it.